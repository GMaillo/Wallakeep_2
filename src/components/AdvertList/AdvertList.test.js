import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import AdvertList from './AdvertList';

configure({ adapter: new Adapter() });

const defaultProps = {
    adverts: {
        count: 1,
        results: [
            {
                createdAt: '2019-10-21T15:26:08.224Z',
                description: 'Vendo XBOX One X como nueva. No tengo tiempo para jugar',
                id: '5db5c80cb00acc1d9437de43',
                name: 'XBOX OneX',
                photo: 'http://localhost:3001/images/anuncios/xbox.jpg',
                price: 35.6,
                tags: ['lifestyle'],
                type: 'sell'
            }
        ],
        success: true
    }
};

const render = props => shallow(<AdvertList {...defaultProps} {...props} />);
let wrapper;


describe('AdvertList component test', () => {
    beforeEach(() => {
        wrapper = render();
    });

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render adverts list container', () => {
        expect(wrapper.find('.adverts-container')).toHaveLength(1);
    });
}); 