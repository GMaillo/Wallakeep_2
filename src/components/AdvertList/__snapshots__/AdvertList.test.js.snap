// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvertList component test should render correctly 1`] = `
<div>
  <div
    className="adverts-container"
  >
    <withRouter(Advert)
      advert={
        Object {
          "createdAt": "2019-10-21T15:26:08.224Z",
          "description": "Vendo XBOX One X como nueva. No tengo tiempo para jugar",
          "id": "5db5c80cb00acc1d9437de43",
          "name": "XBOX OneX",
          "photo": "http://localhost:3001/images/anuncios/xbox.jpg",
          "price": 35.6,
          "tags": Array [
            "lifestyle",
          ],
          "type": "sell",
        }
      }
      key="5db5c80cb00acc1d9437de43"
    />
  </div>
</div>
`;
